### PR TITLE
heroic: fix icon sizes

### LINKED ISSUE #348764: Heroic Games Launcher desktop icon blurry
### Describe the bug

I've noticed that the desktop icon for Heroic Games Launcher is very low resolution. At least that's the case on Gnome, I haven't tried any other desktop environments.

This has been the case for at least the last few months. Hadn't used the application before.

### Steps To Reproduce
1. Install Heroic Games Launcher by adding it to your NixOS or home-manager configuration and switching to it.
2. Search for Heroic and/or add it to your dash.
3. Not a single high res desktop icon to be seen.

### Expected behavior
Seeing the a high res version of the desktop icon.

### Screenshots
Screenshot from 2024-10-15 13-08-37
Screenshot from 2024-10-15 13-09-00

### Notify maintainers

@aidalgol

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.2-cachyos, NixOS, 24.11 (Vicuna), 24.11.20241009.5633bcf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.90.0`
 - nixpkgs: `/nix/store/60sn02zhawl3kwn0r515zff3h6hg6ydz-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Steinhagen**
By default, the 128x128 icon was installed as a 512x512.
The modification uses the svg icon present in the deployment to add all icon sizes + the scalable one.

This should fix: https://github.com/NixOS/nixpkgs/issues/348764

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by aidalgol — CHANGES_REQUESTED** [review]
Thanks for doing this!  There's just some quoting that I'd like changed, and then I'm happy to merge this.

File: pkgs/games/heroic/default.nix [hunk 1]
```diff
@@ -88,9 +90,19 @@ stdenv.mkDerivation (finalAttrs: {
 
     substituteInPlace "$out/share/heroic/flatpak/com.heroicgameslauncher.hgl.desktop" \
       --replace-fail "Exec=heroic-run" "Exec=heroic"
-    mkdir -p "$out/share/applications" "$out/share/icons/hicolor/512x512/apps"
+    mkdir -p "$out/share/applications" "$out/share/icons/hicolor/scalable/apps"
     ln -s "$out/share/heroic/flatpak/com.heroicgameslauncher.hgl.desktop" "$out/share/applications"
-    ln -s "$out/share/heroic/flatpak/com.heroicgameslauncher.hgl.png" "$out/share/icons/hicolor/512x512/apps"
+    install -Dm644 $out/share/heroic/src/frontend/assets/heroic-icon.svg $out/share/icons/hicolor/scalable/apps/com.heroicgameslauncher.hgl.svg
+    for size in 16 22 32 48 64 128 256 512; do
+      dir=$out/share/icons/hicolor/"$size"x"$size"/apps
```
[the remark this answers is not in the document]
Comment (Steinhagen) [reply pkgs/games/heroic/default.nix:97]: This won't work:

       error: undefined variable 'size'
       at /nix/store/i0a3v81pvvmm63sg0rv5an3w53yhz9is-source/pkgs/games/heroic/default.nix:97:39:
           96|     for size in 16 22 32 48 64 128 256 512; do
           97|       dir="$out/share/icons/hicolor/${size}x${size}/apps"

[the remark this answers is not in the document]
Comment (aidalgol) [reply pkgs/games/heroic/default.nix:97]: Oh, of course; this is in a Nix string.  Try this.
```suggestion
      dir="$out/share/icons/hicolor/''${size}x''${size}/apps"
```

[the remark this answers is not in the document]
Comment (Steinhagen) [reply pkgs/games/heroic/default.nix]: I pushed the modifications.

**Review by aidalgol — APPROVED** [review]

File: pkgs/games/heroic/default.nix [hunk 2]
```diff
@@ -88,9 +90,19 @@ stdenv.mkDerivation (finalAttrs: {
...
+    for size in 16 22 32 48 64 128 256 512; do
+      dir="$out/share/icons/hicolor/''${size}x''${size}/apps"
+      mkdir -p $dir
+      rsvg-convert \
+        --keep-aspect-ratio \
+        --width $size \
+        --height $size \
+        --output $dir/com.heroicgameslauncher.hgl.png \
+        $out/share/icons/hicolor/scalable/apps/com.heroicgameslauncher.hgl.svg
+    done
```
Comment (Aleksanaa) [inline pkgs/games/heroic/default.nix:105]: I don't think this is needed if we already included the svg. SVG as desktop icon is part of the freedesktop specification, and can satisfy every display size.

**Review by aidalgol — CHANGES_REQUESTED** [review]

File: pkgs/games/heroic/default.nix (hunk 2 above)
Comment (Steinhagen) [reply pkgs/games/heroic/default.nix]: I did the modifications, but I wasn't able to test it yet.

**Comment by Steinhagen** [comment]
Can someone check if the icon is correctly rendered on Gnome?

**Pull request merged**